core/testsupport/CMakeLists.txt

```diff
@@ -4,10 +4,6 @@
 # higher than kInfo are issued by tests.
 # Stephan Hageboeck, CERN, 2022
 
-if(NOT testing)
```

Comment:
As a consequence is the test library now part of a ROOT install or is it still only part of a build directory?

Reply:
Yes, it's part of the ROOT install. So I guess, the downside of this PR is that it adds 0.5 MB to the install. You think that's acceptable?

Reply:
I am more worried about  adding to our maintenance burden by making something 'new' part of the implicit/de-facto interfaces.  However:
* the files (the header and the .o files) are already distributed if we build the official build with test enabled (I don't know whether we do or not).
* the features is being distributed as a .o files so the user would have to have dig through to find it ... or use `ROOT_ADD_GTEST`

So actually we are technically expanding the proposed interface (adding `ROOT_ADD_GTEST`) but that the point of this PR.

Reply:
Thanks for the assessment!

The `ROOT_ADD_GTEST` seems to be intended as part of the public interface already actually, and you get a warning if the TestSupport library is missing: https://github.com/root-project/root/blob/master/cmake/modules/RootMacros.cmake#L1838

So, it was already possible to use `ROOT_ADD_GTEST` with some ramifications.

Now, this warning will hopefully not happen anymore.